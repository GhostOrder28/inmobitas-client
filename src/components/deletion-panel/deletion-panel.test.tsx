import React from 'react';
import { render, screen } from '@testing-library/react';
import DeletionPanel from './deletion-panel.component';
import '@testing-library/jest-dom'

jest.mock('react-i18next', () => ({
  useTranslation: () => {
    return {
      t: (str: string) => str,
      i18n: {
        changeLanguage: () => new Promise(() => {}),
      },
    };
  },
}));

describe('DeletionPanel', () => {
  const setStateMock = jest.fn();
  const useStateMock: any = (useState: any) => [useState, setStateMock];
  jest.spyOn(React, 'useState').mockImplementation(useStateMock);

  it('should render DatePicker', () => { 
    const wrapper = render(
      <DeletionPanel
        showDeletionMenu={false}
        submitDeletion={() => Promise.resolve()}
        setShowDeletionMenu={useStateMock(false)[0]}
        setMarkedPictures={useStateMock(false)[0]}
      />
    )
    expect(wrapper).toMatchSnapshot();
  });

  it('height should be 0rem', () => {
    render(
      <DeletionPanel
        showDeletionMenu={false}
        submitDeletion={() => Promise.resolve()}
        setShowDeletionMenu={useStateMock(false)[0]}
        setMarkedPictures={useStateMock(false)[0]}
      />
    )
    const deletionPanelContainer = screen.queryByTestId('deletion-panel-container');
    expect(deletionPanelContainer).toHaveStyle({ height: '0rem' })

  })

  it('height should be 2.2rem', () => {
    render(
      <DeletionPanel
        showDeletionMenu={true}
        submitDeletion={() => Promise.resolve()}
        setShowDeletionMenu={useStateMock(false)[0]}
        setMarkedPictures={useStateMock(false)[0]}
      />
    )
    const deletionPanelContainer = screen.queryByTestId('deletion-panel-container');
    expect(deletionPanelContainer).toHaveStyle({ height: '2.2rem' })
  });

  it('should hide the component (height to 0rem)', () => {
    
  });

})

