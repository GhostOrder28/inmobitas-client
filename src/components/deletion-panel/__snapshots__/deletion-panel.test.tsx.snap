// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeletionPanel should render DatePicker 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      data-evergreen-toaster-container=""
    >
      <span
        class="css-1sugtjn"
      />
    </div>
    <div>
      <div
        class="ub-dspl_flex ub-algn-itms_center ub-just-cnt_flex-end ub-top_7-6rem ub-rgt_-8rem ub-w_30prcnt ub-ml_auto ub-h_0rem ub-z-idx_97 ub-ovflw-x_hidden ub-ovflw-y_hidden ub-tstn_1u0x6fi ub-box-szg_border-box"
        data-testid="deletion-panel-container"
      >
        <div
          class="ub-dspl_flex ub-just-cnt_center ub-algn-itms_center ub-w_100prcnt ub-h_100prcnt ub-flx_1 ub-crsr_pointer ub-box-szg_border-box"
        >
          <svg
            class="css-1oxm3o5 ub-w_23px ub-h_23px ub-box-szg_border-box"
            data-icon="tick"
            viewBox="0 0 20 20"
          >
            <path
              d="M17 4c-.28 0-.53.11-.71.29L7 13.59 3.71 10.3A.965.965 0 003 10a1.003 1.003 0 00-.71 1.71l4 4c.18.18.43.29.71.29s.53-.11.71-.29l10-10A1.003 1.003 0 0017 4z"
              fill-rule="evenodd"
            />
          </svg>
        </div>
        <div
          class="ub-dspl_flex ub-just-cnt_center ub-algn-itms_center ub-w_100prcnt ub-h_100prcnt ub-flx_1 ub-crsr_pointer ub-box-szg_border-box"
        >
          <svg
            class="css-1oxm3o5 ub-w_23px ub-h_23px ub-box-szg_border-box"
            data-icon="cross"
            viewBox="0 0 20 20"
          >
            <path
              d="M11.41 10l4.29-4.29c.19-.18.3-.43.3-.71a1.003 1.003 0 00-1.71-.71L10 8.59l-4.29-4.3a1.003 1.003 0 00-1.42 1.42L8.59 10 4.3 14.29c-.19.18-.3.43-.3.71a1.003 1.003 0 001.71.71l4.29-4.3 4.29 4.29c.18.19.43.3.71.3a1.003 1.003 0 00.71-1.71L11.41 10z"
              fill-rule="evenodd"
            />
          </svg>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="ub-dspl_flex ub-algn-itms_center ub-just-cnt_flex-end ub-top_7-6rem ub-rgt_-8rem ub-w_30prcnt ub-ml_auto ub-h_0rem ub-z-idx_97 ub-ovflw-x_hidden ub-ovflw-y_hidden ub-tstn_1u0x6fi ub-box-szg_border-box"
      data-testid="deletion-panel-container"
    >
      <div
        class="ub-dspl_flex ub-just-cnt_center ub-algn-itms_center ub-w_100prcnt ub-h_100prcnt ub-flx_1 ub-crsr_pointer ub-box-szg_border-box"
      >
        <svg
          class="css-1oxm3o5 ub-w_23px ub-h_23px ub-box-szg_border-box"
          data-icon="tick"
          viewBox="0 0 20 20"
        >
          <path
            d="M17 4c-.28 0-.53.11-.71.29L7 13.59 3.71 10.3A.965.965 0 003 10a1.003 1.003 0 00-.71 1.71l4 4c.18.18.43.29.71.29s.53-.11.71-.29l10-10A1.003 1.003 0 0017 4z"
            fill-rule="evenodd"
          />
        </svg>
      </div>
      <div
        class="ub-dspl_flex ub-just-cnt_center ub-algn-itms_center ub-w_100prcnt ub-h_100prcnt ub-flx_1 ub-crsr_pointer ub-box-szg_border-box"
      >
        <svg
          class="css-1oxm3o5 ub-w_23px ub-h_23px ub-box-szg_border-box"
          data-icon="cross"
          viewBox="0 0 20 20"
        >
          <path
            d="M11.41 10l4.29-4.29c.19-.18.3-.43.3-.71a1.003 1.003 0 00-1.71-.71L10 8.59l-4.29-4.3a1.003 1.003 0 00-1.42 1.42L8.59 10 4.3 14.29c-.19.18-.3.43-.3.71a1.003 1.003 0 001.71.71l4.29-4.3 4.29 4.29c.18.19.43.3.71.3a1.003 1.003 0 00.71-1.71L11.41 10z"
            fill-rule="evenodd"
          />
        </svg>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
